core/src/main/java/org/apache/calcite/rel/core/Project.java

```diff
@@ -145,6 +146,12 @@ public RelNode accept(RexShuttle shuttle) {
     if (this.exps == exps) {
       return this;
     }
+    final RelDataType rowType =
+        RexUtil.createStructType(
+            getInput().getCluster().getTypeFactory(),
+            exps,
+            this.rowType.getFieldNames(),
+            SqlValidatorUtil.F_SUGGESTER);
     return copy(traitSet, getInput(), exps, rowType);
```

Comment:
Do we need a `SqlValidatorUtil.Suggester` here ? What's the problem if we keep the original fields names(if there are some same name fields.)

Reply:
@danny0405 , you're right, we should not need a Suggester here. I'll change the parameter to `null`.